File: app/src/main/java/com/hana053/micropost/interactors/FeedInteractor.java
package com.hana053.micropost.interactors;

import com.hana053.micropost.domain.Micropost;

import java.util.List;

import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

public interface FeedInteractor {

    @GET("feed?count=20")
    Observable<List<Micropost>> loadNextFeed(@Query("sinceId") Long sinceId);

    @GET("feed?count=20")
    Observable<List<Micropost>> loadPrevFeed(@Query("maxId") Long maxId);
}

File: app/src/main/java/com/hana053/micropost/interactor/FeedInteractor.kt
package com.hana053.myapp.interactors

import com.hana053.micropost.domain.Micropost
import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable

interface FeedInteractor {

    @GET("feed?count=20")
    fun loadNextFeed(@Query("sinceId") sinceId: Long?): Observable<List<Micropost>>

    @GET("feed?count=20")
    fun loadPrevFeed(@Query("maxId") maxId: Long?): Observable<List<Micropost>>
}